getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,160 @@
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+                .addFeatures(feat)
+                .setImage(img)
+                .build()
+
+        requests.add(request)
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(requests)
+        val responses = response.responsesList
+        var output = "" //kind of log
+
+        for (resp in responses) {
+            if (resp.hasError()) {
+                output += "Error: ${resp.error.message} \n"
+                return output
+            }
+
+            downloadObject(objectName) //image is now downloaded to /tmp/<objectName>
+            val imgBuff = ImageIO.read(Paths.get("/tmp/$objectName").toFile())
+            val gfx = imgBuff.createGraphics()
+            val extension = (Paths.get("/tmp/$objectName").toFile().extension)
+
+            for (annotation in resp.faceAnnotationsList) {
+                val emoji = bestEmoji(annotation)
+                val imgEmoji = ImageIO.read(Paths.get(EMOJIS_PATH + emojiPic[emoji]).toFile())
+
+                output += """
+                    joy: ${annotation.joyLikelihood}
+                    anger: ${annotation.angerLikelihood}
+                    surprise: ${annotation.surpriseLikelihood}
+                    sorrow: ${annotation.sorrowLikelihood}
+                    position: ${annotation.boundingPoly}
+                """
+                output+= "EMOJI DETECTED: $emoji \n"
+                output+= "****************************\n"
+
+                val poly = Polygon()
+                for (vertex in annotation.fdBoundingPoly.verticesList) {
+                    poly.addPoint(vertex.x, vertex.y)
+                }
+                /* This draws a polygon around the detected face
+                gfx.stroke = BasicStroke(5f)
+                gfx.color = Color(0x00ff00)
+                gfx.draw(poly)
+                */
+                output += "${poly.xpoints[0]}, ${poly.ypoints[0]}, ${poly.xpoints[1]}, ${poly.ypoints[2]}"
+                val height = poly.ypoints[2] - poly.ypoints[0]
+                val width = poly.xpoints[1] - poly.xpoints[0]
+                //Draws emoji on detected face
+                gfx.drawImage(imgEmoji, poly.xpoints[0], poly.ypoints[1], height, width, null)
+            }
+            //Writing emojified image to temp file
+            val writeTo = "/tmp/emojify-$objectName"
+            ImageIO.write(imgBuff, extension, Paths.get(writeTo).toFile());
+
+            //Uploading emojified image to GCS
+            val blob = bucket.create("emojified/emojified-$objectName",  Files.readAllBytes(Paths.get(writeTo)))
+
+            //Making it public
+            blob.createAcl(Acl.of(Acl.User.ofAllUsers(), Acl.Role.READER))
+        }
+        output+="Done!"
+        //Everything went well; we can return the public url!
+        return publicUrl
```

Comment:
Let's return an object here. Can be a data object with just one property, `uri`. But this would be better in case we want to add properties later on.